### DISCUSSION TITLE
Should we use a different library to avoid `eval()` or get rid of the function?

### DISCUSSION THREAD

**Discussion by pybash1** [pybash1/py_everything#64] [General]
@Morgan-Phoenix as mentioned in #62 we should be avoiding the use of `eval()` function but to avoid it we either have to remove the function entirely or find a new way for evaluating the expressions. So I found this library should we use it? 

Library name - [py-expression-eval](https://pypi.org/project/py-expression-eval/)

The library is good indeed but the problem is that it creates a new dependency, and it would slow down code. So what do you think? The library's overall speed wouldn't drop that much but it definitely will drop. And speed is a great factor in python code.

**Comment by suy0x1** [comment]
We can use a built in library `ast`

**Reply by pybash1** [reply]
Yes, but the library isnt intended to that, it is for Abstract Syntax Trees?

**Reply by suy0x1** [reply]
Well if u want you can use it like that

**Reply by pybash1** [reply]
What do you think about `py-expression-eval`? @Morgan-Phoenix

**Reply by suy0x1** [reply]
I'm thinking not to implement it.
Reasons:
* `ast` is already there
*  no good reason to slow down code
*  no good reason to increase dependency count

**Reply by pybash1** [reply]
Yes, same thoughts. Could you make it with `ast`, because I do not have experience with the `ast` library and would take time to learn it?

**Reply by suy0x1** [reply]
Check out line 131, 143, 144
https://github.com/Morgan-Phoenix/EnroCrypt/blob/main/enrocrypt/encryption.py

Use cases

**Reply by pybash1** [reply]
Ok.

**Reply by pybash1** [reply]
Ok, looks, good, so can you implememnt it?

**Reply by suy0x1** [reply]
Not today

**Reply by pybash1** [reply]
Yes whenever you get time not a problem.

**Reply by suy0x1** [reply]
But you know it's a security issue and it's good that we fix it now but if you have no problem than I will do it tomorrow

**Reply by pybash1** [reply]
Yes ok, tomorrow it is then.

**Reply by pybash1** [reply]
I understand it is a security issue but if it has been there for this much time, how much harm would one more day do?

**Reply by suy0x1** [reply]
Ok 
I'm trying on phone

**Reply by pybash1** [reply]
Its ok, you can do it tomorrow there isn't a hurry.
